openedx/core/djangoapps/content_tagging/handlers.py

```diff
@@ -125,3 +128,19 @@ def delete_tag_library_block(**kwargs):
         delete_library_block_tags(str(library_block_data.usage_key))
     except Exception as err:  # pylint: disable=broad-except
         log.error(f"Failed to delete library block tags: {err}")
+
+
+@receiver(XBLOCK_DUPLICATED)
+def duplicate_tags(**kwargs):
+    """
+    Duplicates tags associated with an XBlock whenever the block is duplicated to a new location.
+    """
+    xblock_data = kwargs.get("xblock_info", None)
+    if not xblock_data or not isinstance(xblock_data, DuplicatedXBlockData):
+        log.error("Received null or incorrect data for event")
+        return
+
+    copy_object_tags(
+        xblock_data.source_usage_key,
+        xblock_data.usage_key,
+    )
```

Comment:
Currently our clipboard code is not async because it seems fast enough. I don't think we need to make this async unless we find that it's slow for some reason.